pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -233,6 +233,7 @@ private int storeValues(final Object[] arr, int elementOid, final int[] dims, in
         int len = ByteConverter.int4(fieldBytes, pos);
         pos += 4;
         if (len == -1) {
+          arr[i] = null;
```

Comment:
Just a side note: Java arrays are guaranteed to be initialized with `null` contents, so this write might be excessive.
On the other hand, it does not hurt

Reply:
I specifically asked for this for readability, assuming the compiler would optimize it out

Reply:
TL;DR: I meant to add a comment so the one who finds "excessive code" would refrain from deleting the line :)

Reply:
As I understand it, the javac compiler is unlikely to get rid of it. The server optimizer/compiler /might/ be able to optimize it away. 

The nature of this being a recursive method could make it difficult for the jvm to realize it is not actually necessary work.

Reply:
I have added a comment indicating that the setting to null is for clarity of the reader.